### PR TITLE
Add breadcrumb story

### LINKED ISSUE #385: Add Breadcrumb component as per the designs.
**Description**

This component is widely used in the new designs for the websites. So, it is better to create it here.

**Problems solved**

A “breadcrumb” is a type of secondary navigation scheme that reveals the user's location on a website or Web application. It helps us to provide a better experience and is widely used in the new designs. 
image

**Want to take up?**

[Yes]

### PR DESCRIPTION
**Pull request by Dhruvi16**
## Fixes

Fixes #385  by @Dhruvi16 

## Description

Adds the breadcrumb component as per the design library.

## Screenshots

image

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by Dhruvi16** [comment]
I am not sure about the icon I used as the item separator, @hugosolar @dhruvkb could you please help me figure that out?
Thanks

**Comment by dhruvkb** [comment]
@Dhruvi16 I think we can use either the chevron or the caret from [the icon font](https://cc-fonts.netlify.app/?path=/docs/vocabulary-icons--page#arrows). @panchovm can help us decide.

**Comment by fcoveram** [comment]
The component should have the icon, I am using `chevron-right`. I will add the component to the Design Library.

File: src/styles/breadcrumb.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
+$breadcrumb-item-color: $color-tomato;
+$breadcrumb-item-active-color: $color-dark-slate-gray;
+$breadcrumb-item-hover-color: $color-tomato;
+$breadcrumb-item-padding-horizontal: $space-small;
+$breadcrumb-item-separator-color: $color-gray;
+
+
```
Comment (dhruvkb) [inline src/styles/breadcrumb.scss]: Change to a single line.

File: src/styles/breadcrumb.scss [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@import "bulma/sass/components/breadcrumb.sass";
+
+
```
Comment (dhruvkb) [inline src/styles/breadcrumb.scss]: Change to a single line.

File: src/styles/breadcrumb.scss [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+nav.breadcrumb > ul {
+
+  font-style: normal;
+  font-weight: 600;
+  font-size: 13px;
```
Comment (dhruvkb) [inline src/styles/breadcrumb.scss]: Prefer using tokens such as `$font-size-caption`.

File: src/styles/breadcrumb.scss [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  line-height: 19px;
```
Comment (dhruvkb) [inline src/styles/breadcrumb.scss]: Use line heights as fraction of font-size so 19/13 = 1.4615384615.
```suggestion
  line-height: 1.4615384615;
```

@panchovm this is not the same as the other line heights which all use a constant ratio. Is that intentional?

File: src/styles/breadcrumb.scss [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  li+li:before {
+    font-family: "Vocabulary Icons";
+    content: "\e901";
+  };
+
+  li:hover {
+    text-underline-position: below;
+  };
```
Comment (dhruvkb) [inline src/styles/breadcrumb.scss]: Scopes are never ended with semicolons.

File: src/styles/breadcrumb.scss [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
```
Comment (dhruvkb) [inline src/styles/breadcrumb.scss]: Remove stray newlines at the start and end of blocks.

File: src/stories/breadcrumb.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import { Meta, Preview, Story } from '@storybook/addon-docs/blocks'
+import { figmaConfig } from './helpers'
+
+<Meta title="Patterns/Breadcrumb" />
+
+# Breadcrumb
+
+export const Breadcrumb = () => `
```
Comment (dhruvkb) [inline src/stories/breadcrumb.stories.mdx]: Functions must follow lowerCamelCase.

File: src/stories/breadcrumb.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+`
+
+The Breadcrumb has two text styles: links to parent pages and the page the user is seeing. The parent links should be corresponding to the site map defined. At the same time, the color used for links should be the main color for links across the site.
+
+For accessibility purposes, the component should be placed above the page's title.
+
+<Preview mdxSource={Breadcrumb()}>
+  <Story name="Breadcrumb" parameters={{
+    design: figmaConfig('6368%3A9')}}>{Breadcrumb()}</Story>
+</Preview>
```
Comment (dhruvkb) [inline src/stories/breadcrumb.stories.mdx]: - If there is only one story, name it Default. 
- Pay close attention to line breaks and indentation.

```suggestion
<Preview mdxSource={breadcrumb()}>
  <Story name="Default" parameters={{
    design: figmaConfig('6368%3A9')
  }}>{
    breadcrumb()
  }</Story>
</Preview>
```

**Review by dhruvkb — CHANGES_REQUESTED** [review]
While most of these comments will seem nitpicking, please note that this is just to imbibe in you the code style that is being followed in the project.

File: src/styles/breadcrumb.scss (hunk 3 above)
Comment (Dhruvi16) [reply src/styles/breadcrumb.scss]: I used caption but it gave me a different result than expected. Like font-weight was not same.

File: src/stories/breadcrumb.stories.mdx (hunk 8 above)
Comment (Dhruvi16) [reply src/stories/breadcrumb.stories.mdx]: Can you suggest any particular extension that can do this for me or the one you use for maintaining indentation?
Thanks in advance

File: src/styles/breadcrumb.scss (hunk 3 above)
Comment (Dhruvi16) [reply src/styles/breadcrumb.scss]: image

Comment (dhruvkb) [reply src/styles/breadcrumb.scss]: `$font-size-caption` is just a font-size.
```scss
font-size: $font-size-caption;
```

File: src/stories/breadcrumb.stories.mdx (hunk 8 above)
Comment (dhruvkb) [reply src/stories/breadcrumb.stories.mdx]: I use [WebStorm](https://www.jetbrains.com/webstorm/) as my IDE (paid software, free version if you have an .edu or .ac email address from your institute) and simultaneously keep an eye out for this manually too. You can setup similar tools in Visual Studio Code too.

File: src/styles/breadcrumb.scss (hunk 4 above)
Comment (Dhruvi16) [reply src/styles/breadcrumb.scss]: If that helps -
font-family: Source Sans Pro;
font-style: normal;
font-weight: 600;
font-size: 13px;
line-height: 19px;
This is in the design library.

**Review by dhruvkb — APPROVED** [review]

**Pull request merged**